HealthcareSystem/EventSourcingService/Service/PatientSchedulingEventService/EventStorePatientEndSchedulingService.cs

```diff
@@ -0,0 +1,219 @@
+                                                   (e => e.ReasonForEndOfAppointment == ReasonForEndOfAppointment.Success);
+
+                successfulSchedulingGenderStatistic.NumberOfWomen = successfulScheduling.Where(e => e.UserGender == Gender.Female).Count();
+                successfulSchedulingGenderStatistic.NumberOfMen = successfulScheduling.Where(e => e.UserGender == Gender.Male).Count();
+                successfulSchedulingGenderStatistic.TotalNumber = successfulScheduling.Count();
+            }
+            catch (Exception)
+            {
+                return new GenderStatisticDTO();
+            }
+
+            return successfulSchedulingGenderStatistic;
+        }
+
+        public MinAvgMaxStatisticDTO SuccessfulSchedulingAgeStatistic()
+        {
+            MinAvgMaxStatisticDTO successfulSchedulingAgeStatistic = new MinAvgMaxStatisticDTO();
+
+            try
+            {
+                IEnumerable<PatientEndSchedulingEvent> successfulScheduling = _patientEndSchedulingEventRepository.GetAll
+                                                   (e => e.ReasonForEndOfAppointment == ReasonForEndOfAppointment.Success);
+
+                successfulSchedulingAgeStatistic.Minimum = successfulScheduling.Min(e => e.UserAge);
+                successfulSchedulingAgeStatistic.Average = (int)successfulScheduling.Average(e => e.UserAge);
+                successfulSchedulingAgeStatistic.Maximum = successfulScheduling.Max(e => e.UserAge);
+            }
+            catch (Exception)
+            {
+                return new MinAvgMaxStatisticDTO();
+            }
+
+            return successfulSchedulingAgeStatistic;
+        }
+
+        public AverageDurationDTO SuccessfulSchedulingAgeDuration()
+        {
+            AverageDurationDTO averageDTO = new AverageDurationDTO();
+
+            try
+            {
+                int averageAge = (int)_patientEndSchedulingEventRepository.GetAll().Average(e => e.UserAge);
+
+                IEnumerable<PatientEndSchedulingEvent> successfulScheduling = _patientEndSchedulingEventRepository.GetAll
+                                                       (e => e.ReasonForEndOfAppointment == ReasonForEndOfAppointment.Success);
+
+                IEnumerable<TimeSpan> youthDuration = successfulScheduling.Where(e => e.UserAge < averageAge)
+                                                      .Select(e => e.TriggerTime - e.StartSchedulingEventTime);
+                IEnumerable<TimeSpan> elderDuration = successfulScheduling.Where(e => e.UserAge >= averageAge)
+                                                      .Select(e => e.TriggerTime - e.StartSchedulingEventTime);
+
+                averageDTO.DurationFirst = (int)youthDuration.DefaultIfEmpty().Average(t => t.TotalMinutes);
+                averageDTO.DurationSecond = (int)elderDuration.DefaultIfEmpty().Average(t => t.TotalMinutes);
```

Comment:
Something that stands out to me after testing this out myself - minutes are a bad metric for the duration, especially if you round them out. It generally takes me less than a minute to schedule an appointment. You should probably switch to seconds.